templates/project-ts/src/Add.ts

```diff
@@ -1,25 +1,22 @@
 import { Field, SmartContract, state, State, method } from 'o1js';
+import { AddProgramProof } from './AddZKprogram.js';
 
 /**
  * Basic Example
  * See https://docs.minaprotocol.com/zkapps for more info.
  *
- * The Add contract initializes the state variable 'num' to be a Field(1) value by default when deployed.
- * When the 'update' method is called, the Add contract adds Field(2) to its 'num' contract state.
+ * The Add contract verifies a ZKprogram proof and updates a 'num' state variable.
+ * When the 'settleState' method is called, the Add contract verifies a
+ * proof from the 'AddZKprogram' and saves the 'num' value to the contract state.
  *
  * This file is safe to delete and replace with your own contract.
  */
 export class Add extends SmartContract {
   @state(Field) num = State<Field>();
 
-  init() {
-    super.init();
-    this.num.set(Field(1));
-  }
-
-  @method async update() {
-    const currentState = this.num.getAndRequireEquals();
-    const newState = currentState.add(2);
-    this.num.set(newState);
+  @method async settleState(proof: AddProgramProof) {
+    proof.verify();
```

Comment:
We should add a precondition that the input of the proof is the current zkapp state

Reply:
I added a precondition and redeployed the contract.